src/components/FileTree.tsx

```diff
@@ -105,6 +106,7 @@ const FileTreeItem = ({
 
   function openFile() {
     if (fileOrDir.children !== undefined) return // Don't open directories
+    kclManager.setCode('')
```

Comment:
Similarly nervous about using `kclManager` here, but the problem I was having is that when I would navigate to new files that were supposed to be empty from non-empty files, the code in the text editor wouldn't update. This clearing out of the current code seemed to do the trick.

Reply:
Right, there is some logic around empty files with the kclManager that you probably hit, like we don't do an execution if the file is empty, but maybe the assumption is that is if the file is empty it's coming from CodeMirror such that when it becomes empty from elsewhere CodeMirror doesn't get the prompt, just a theory, this is fine though.